### PR TITLE
[main] Upgrade Python3 to 3.9.9

### PR DESCRIPTION
**Pull request by thcrain-msft**
###### Merge Checklist  

**All** boxes should be checked before merging the PR *(just tick any boxes which don't apply to this PR)*
- [x] The toolchain has been rebuilt successfully (or no changes were made to it)
- [x] The toolchain/worker package manifests are up-to-date
- [x] Any updated packages successfully build (or no packages were changed)
- [x] Package tests (%check section) have been verified with RUN_CHECK=y for existing SPEC files, or added to new SPEC files
- [x] All package sources are available
- [x] cgmanifest files are up-to-date and sorted (`./cgmanifest.json`, `./toolkit/tools/cgmanifest.json`, `./toolkit/scripts/toolchain/cgmanifest.json`, `.github/workflows/cgmanifest.json`)
- [x] LICENSE-MAP files are up-to-date (`./SPECS/LICENSES-AND-NOTICES/data/licenses.json`, `./SPECS/LICENSES-AND-NOTICES/LICENSES-MAP.md`, `./SPECS/LICENSES-AND-NOTICES/LICENSE-EXCEPTIONS.PHOTON`)
- [x] All source files have up-to-date hashes in the `*.signatures.json` files
- [x] `sudo make go-tidy-all` and `sudo make go-test-coverage` pass
- [x] Documentation has been updated to match any changes to the build system
- [x] Ready to merge

---

###### Summary 

Upgrade Python 3 to 3.9.9 as part of efforts to keep our toolchain modern and supported.

###### Change Log  

- Python 3 upgraded to latest 3.9.X version in raw toolchain and official toolchain
- python3-libs subpackage has absorbed the python3-xml subpackage
- Where necessary, easy_install3 usage was replaced with pip3 in check sections 
- Packages with pre-built Cython C sources have been changed to regenerate from source
- Several packages had patches applied to enable building with Python 3.9
- The python-nose package has been removed due to upstream abandonment

###### Does this affect the toolchain?  

**YES**

###### Test Methodology

- Local and pipeline full builds

### CODE REVIEW

File: SPECS/python-gevent/python-gevent.spec [hunk 1]
```diff
@@ -40,6 +41,7 @@ Features include:
 
 %prep
 %autosetup -p 1 -n gevent-%{version}
+rm $(grep -rl '/\* Generated by Cython')
```
Comment (PawelWMS) [inline SPECS/python-gevent/python-gevent.spec:44]: Nit: could use a comment, so people know why we're doing this.

File: SPECS/python-markupsafe/python-markupsafe.spec [hunk 2]
```diff
@@ -20,7 +20,7 @@ BuildRequires:  python3-setuptools
 BuildRequires:  python3-xml
 Requires:       python3
 Provides:       python3dist(markupsafe) = %{version}-%{release}
-Provides:       python3.7dist(markupsafe) = %{version}-%{release}
+Provides:       python3.9dist(markupsafe) = %{version}-%{release}
```
Comment (PawelWMS) [inline SPECS/python-markupsafe/python-markupsafe.spec:23]: Suggestion: do you think we could use a macro here, which would always extract the version number from `%{python3_version}` and place that number here, so we don't miss any future updates?

File: SPECS/python3/python3.spec [hunk 3]
```diff
@@ -1,19 +1,21 @@
 %global openssl_flags -DOPENSSL_NO_SSL3 -DOPENSSL_NO_SSL2 -DOPENSSL_NO_COMP
 %global __brp_python_bytecompile %{nil}
+%global majmin 3.9
```
Comment (PawelWMS) [inline SPECS/python3/python3.spec:3]: Might be easier for future updates:

```suggestion
%global majmin %(echo %{version} | cut -d. -f1-2)
```

File: SPECS/python3/python3.spec [hunk 4]
```diff
@@ -112,10 +103,9 @@ to build python programs.
 %package        pip
 Summary:        The PyPA recommended tool for installing Python packages.
 Group:          Development/Tools
-Requires:       python3 = %{version}-%{release}
-Requires:       python3-xml = %{version}-%{release}
+Requires:       %{name} = %{version}-%{release}
 Provides:       python3dist(pip) = %{version}-%{release}
-Provides:       python3.7dist(pip) = %{version}-%{release}
+Provides:       python3.9dist(pip) = %{version}-%{release}
```
Comment (PawelWMS) [inline SPECS/python3/python3.spec:108]: Can/Should we use `%{majmin}` here?

```suggestion
Provides:       python%{majmin}dist(pip) = %{version}-%{release}
```

File: SPECS/python3/python3.spec (hunk 3 above)
Comment (thcrain-msft) [reply SPECS/python3/python3.spec:3]: Won't fix- I did try this early on, but surprisingly it failed to cut the version down to maj.min! Possibly a future issue to fix with regards to toolchain builds.

File: toolkit/scripts/toolchain/container/toolchain_build_in_chroot.sh [hunk 5]
```diff
@@ -781,22 +781,23 @@ popd
...
-pushd Python-3.7.4
+echo Python-3.9.9
+tar xf Python-3.9.9.tar.xz
+pushd Python-3.9.9
 ./configure --prefix=/usr       \
+            --with-platlibdir=lib \
             --enable-shared     \
             --with-system-expat \
             --with-system-ffi   \
             --with-ensurepip=yes
```
Comment (PawelWMS) [inline toolkit/scripts/toolchain/container/toolchain_build_in_chroot.sh:792]: Not sure it makes a difference, but I think you switched this to `no` in the spec.

File: SPECS/python-gevent/python-gevent.spec (hunk 1 above)
Comment (thcrain-msft) [reply SPECS/python-gevent/python-gevent.spec:45]: Added, thanks

File: SPECS/python3/python3.spec (hunk 4 above)
Comment (thcrain-msft) [reply SPECS/python3/python3.spec]: Suggestion commit taken, thanks!

File: toolkit/scripts/toolchain/container/toolchain_build_in_chroot.sh (hunk 5 above)
Comment (thcrain-msft) [reply toolkit/scripts/toolchain/container/toolchain_build_in_chroot.sh:792]: That's intentional, the `ensurepip` process works fine here but fails in RPM build environment (there should be some links for context in the spec).

**Pull request merged**